src/ml-engine/main_app/models/black_litterman/BlPortfolioModel.py

```
@@ -0,0 +1,108 @@
+        omega = np.diag((1 - np.diagonal(confidence) + 0.05))  # add small floor for stability
+
+        # Step 3: Apply Black-Litterman model
+        picking_matrix = [v.Weights for v in views] # picking matrix of weights
+        return_vector = np.array([v.ExpectedReturn for v in views])
+        bl = BlackLittermanModel(S, pi=prior, omega=omega, P=picking_matrix, Q=return_vector)
+        bl_return = bl.bl_returns()
+        bl_cov = bl.bl_cov()
+
+        # Step 4: Get portfolio weights
+        ef = EfficientFrontier(bl_return, bl_cov)
+        weights = ef.max_sharpe()  # Uncomment this or choose another optimization objective
```

Ruff (0.11.9)

Local variable `weights` is assigned to but never used

Remove assignment to unused variable `weights`

(F841)